File: patch_sitnovate.py
import os

path = r"public/sitnovate-app/assets/index-CFGrlOPU.js"

if not os.path.exists(path):
    print(f"Error: File not found at {path}")
    exit(1)

with open(path, 'rb') as f:
    content = f.read()

# Target string part: The end of Jash Chauhan's entry
# We know from debug that it ends with the linkedin url and a closing brace.
search_bytes = b'linkedin:"https://www.linkedin.com/in/jash-chauhan-2073a0294/"}'

if search_bytes not in content:
    print("Exact match not found. Debugging context...")
    # Try finding the linkedin URL part only
    url_part = b'in/jash-chauhan-2073a0294/'
    idx = content.find(url_part)
    if idx != -1:
         print(f"Found URL part at {idx}. Context around it:")
         print(content[idx:idx+100])
    exit(1)

print(f"Found match at index: {content.find(search_bytes)}")

# Construct replacement
# We want to keep the matched part (end of Jash's object) and append the new objects.
# Note: Using literal newline \n for the backtick strings, matching the file's style.

new_members_bytes = (
    b',{'
    b'text:`Saksham Wadhankar\nWeb Developer`,'
    b'image:"/sitnovate-app/assets/images/Team/sakshamwadhankar.webp",'
    b'linkedin:"https://www.linkedin.com/in/saksham-wadhankar"'
    b'},{'
    b'text:`Om Rai\nWeb Developer`,'
    b'image:"/sitnovate-app/assets/images/Team/OmRai.webp",'
    b'linkedin:"https://www.linkedin.com/in/om-rai-07a368377/"'
    b'}'
)

# Replace occurrence
replacement = search_bytes + new_members_bytes
new_content = content.replace(search_bytes, replacement)

with open(path, 'wb') as f:
    f.write(new_content)

print("Patch applied successfully.")

File: patch_sitnovate.js
const fs = require('fs');
const path = 'public/sitnovate-app/assets/index-CFGrlOPU.js';

try {
    let content = fs.readFileSync(path, 'utf8');
    const name = 'Jash Chauhan';
    const idx = content.indexOf(name);

    if (idx === -1) {
        console.error('Jash Chauhan not found!');
        process.exit(1);
    }

    console.log(`Found '${name}' at ${idx}.`);

    // Show context for debugging
    const snippet = content.slice(idx, idx + 400);
    console.log('CONTEXT START:', snippet);

    // Target string: ends with closing brace of the object
    const targetStr = 'linkedin:"https://www.linkedin.com/in/jash-chauhan-2073a0294/"}';
    const targetIdx = content.indexOf(targetStr, idx);

    if (targetIdx === -1) {
        console.error('Exact target string for replacement not found.');

        // Debugging: try to find just the ID part to see what is different
        const partial = '2073a0294';
        const partialIdx = content.indexOf(partial, idx);
        if (partialIdx !== -1) {
            console.log('Found partial ID at', partialIdx);
            console.log('Surrounding text:', content.slice(partialIdx - 50, partialIdx + 50));
        }
        process.exit(1);
    }

    console.log(`Found target string at ${targetIdx}. Applying patch...`);

    // New members to append
    const newMembers = `,{text:\`Saksham Wadhankar\nWeb Developer\`,image:"/sitnovate-app/assets/images/Team/sakshamwadhankar.webp",linkedin:"https://www.linkedin.com/in/saksham-wadhankar"},{text:\`Om Rai\nWeb Developer\`,image:"/sitnovate-app/assets/images/Team/OmRai.webp",linkedin:"https://www.linkedin.com/in/om-rai-07a368377/"}`;

    const replacement = targetStr + newMembers;

    // Use substring replacement to avoid regex special char issues or replaceAll behavior
    const before = content.slice(0, targetIdx);
    const after = content.slice(targetIdx + targetStr.length);
    const newContent = before + replacement + after;

    fs.writeFileSync(path, newContent, 'utf8');
    console.log('Successfully patched.');

} catch (e) {
    console.error('Error:', e);
}
